src/components/PostDetails.tsx

```diff
@@ -1,117 +1,129 @@
+    if (selectedPost) {
+      deleteCommentFromState(commentId);
+      deleteComments(commentId)
+        .then(() => {
+          getComments(selectedPost.id)
+            .then(commentsFromServer => {
+              handleSetComments(commentsFromServer);
+            });
+        });
+    }
+  };
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${selectedPost?.id}: ${selectedPost?.title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {selectedPost?.body}
           </p>
         </div>
 
         <div className="block">
-          <Loader />
-
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
-
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
-
-          <p className="title is-4">Comments:</p>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+          {isLoadingComments && <Loader />}
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
+          {ErrorMessage === Errors.CommentApi && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              {Errors.CommentApi}
             </div>
-          </article>
+          )}
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a
-                href="mailto:misha@mate.academy"
-                data-cy="CommentAuthor"
-              >
-                Misha Hrynko
-              </a>
-
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div
-              className="message-body"
-              data-cy="CommentBody"
+          {!comments?.length
+            && !isLoadingComments
+            && !ErrorMessage.length
+            && (
```

Comment:
whoa, quite messy here. I encourage you to think of a way of simplifying this one